### PR TITLE
Laboratory work #4, Chekmareva Kristina - 21FPL1

### PR DESCRIPTION
**Pull request by kschekmareva**

### CODE REVIEW

**Comment by artyomtugaryov** [comment]
Nothing to review(

File: lab_1_keywords_tfidf/start.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
     if tokenization:
         no_stop_words = remove_stop_words(tokenization, stop_words)
 
+
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/start.py:49]: revert

File: lab_3_keywords_textrank/main.py [hunk 2]
```diff
@@ -978,7 +978,7 @@ def run(self) -> Optional[dict[str, dict[str, float]]]:
             keywords_path = self._materials_path / f'{self.themes.index(theme)}_keywords.txt'
 
             with (open(text_path, 'r', encoding='utf-8') as text_to_read,
-                    open(keywords_path, 'r', encoding='utf-8') as keywords_to_read):
+                           open(keywords_path, 'r', encoding='utf-8') as keywords_to_read):
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:981]: revert

File: lab_3_keywords_textrank/start.py [hunk 3]
```diff
@@ -17,7 +17,8 @@
     KeywordExtractionBenchmark
 )
 
-
+from lab_3_keywords_textrank.main import extract_pairs, TextPreprocessor, \
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/start.py:20]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_summarization_textrank/main.py [hunk 4]
```diff
@@ -146,10 +210,17 @@ def calculate_similarity(sequence: Union[list, tuple], other_sequence: Union[lis
...
+    if not sequence or not sequence:
+        return 0
+    intersection = list(set(sequence) & set(other_sequence))
+    unification = list(set().union(sequence, other_sequence))
+    measure_of_jacquard = float(len(intersection) / len(unification))
+    return measure_of_jacquard
 
 
-class SimilarityMatrix:
+class SimilarityMatrix():
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:223]: ?revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_summarization_textrank/main.py [hunk 5]
```diff
@@ -146,10 +209,17 @@ def calculate_similarity(sequence: Union[list, tuple], other_sequence: Union[lis
     :param other_sequence: a sequence of items
     :return: similarity score
     """
-    pass
+    if not isinstance(sequence, (list, tuple)) or not isinstance(other_sequence, (list, tuple)):
+        raise ValueError
+    if not sequence or not sequence:
+        return 0
+    intersection = list(set(sequence) & set(other_sequence))
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:216]: what is this operation about?

File: lab_4_summarization_textrank/main.py [hunk 6]
```diff
@@ -160,22 +230,29 @@ def __init__(self) -> None:
...
     def calculate_inout_score(self, vertex: Sentence) -> int:
         """
         Retrieves a number of vertices that are similar (i.e. have similarity score > 0) to the input one
         :param vertex
         :return:
         """
-        pass
+        if not isinstance(vertex, Sentence):
+            raise ValueError
+        summa = 0
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:251]: rename to something not translit

**Review by demid5111 — CHANGES_REQUESTED** [review]

File: lab_4_summarization_textrank/start.py [hunk 7]
```diff
@@ -26,7 +30,20 @@
...
+    completed_sentences = preprocess_sentences.get_sentences(text)
+    encoder = SentenceEncoder()
+    encoder.encode_sentences(completed_sentences)
+    for one_sentence in completed_sentences:
+        print(*(one_sentence.get_encoded()))
+    matrix_graph = SimilarityMatrix()
+    matrix_graph.fill_from_sentences(completed_sentences)
+    text_rank = TextRankSummarizer(matrix_graph)
+    text_rank.train()
+    COMPLETED_SUMMARY = text_rank.make_summary(5)
```
Comment (demid5111) [inline lab_4_summarization_textrank/start.py:44]: summary not longer than 20% of text

10 -> 2

**Pull request closed without merging**